FinSearchUnified/Bundle/SearchBundleFindologic/ConditionHandler/PriceConditionHandler.php

```diff
@@ -27,13 +27,16 @@ public function supportsCondition(ConditionInterface $condition)
      * Handles the passed condition object.
      *
      * @param ConditionInterface $condition
-     * @param QueryBuilder $query
+     * @param NewQueryBuilder $query
      * @param ShopContextInterface $context
      *
      * @throws Exception
      */
-    public function generateCondition(ConditionInterface $condition, QueryBuilder $query, ShopContextInterface $context)
-    {
+    public function generateCondition(
+        ConditionInterface $condition,
+        NewQueryBuilder $query,
```

Comment:
I'd suggest using the interface here.

Reply:
We do not have any interface for this. We have an abstract class, which is being used here. (Applicable to other comments below as well)